app/src/test/kotlin/beam/starter/AppTest.kt

```diff
@@ -0,0 +1,28 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// https://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+package beam.starter
+
+import org.apache.beam.sdk.testing.PAssert
+import org.apache.beam.sdk.testing.TestPipeline
+import org.junit.Rule
```

Comment:
Is junit idiomatic for Kotlin?

Reply:
If we want to use `TestPipeline` it has to be a `@Rule` which is only supported on JUnit 4 (JUnit 5 is not supported). I guess we could use a "normal" pipeline and use Kotlin's tests.

Reply:
For my example presented in Beam Summit, I also used `TestPipeline` with a Junit 4 `@Rule` :

https://github.com/tosun-si/teams-league-kotlin-dlq-asgarde-beam-summit/blob/main/src/test/kotlin/fr/groupbees/domain_ptransform/TeamStatsTransformTest.kt